<!doctype html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Sequence Alignment Pipeline</title>
  <meta name="description" content="">
  <meta name="author" content="Mukarram Hossain and Simon Frost">
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
  <link rel="stylesheet" href="libraries/frameworks/revealjs/css/reveal.min.css">
  <link rel="stylesheet" href="libraries/frameworks/revealjs/css/theme/serif.css" id="theme">
  <link rel="stylesheet" href="libraries/highlighters/highlight.js/css/tomorrow.css" id="theme">
  <!--[if lt IE 9]>
  <script src="lib/js/html5shiv.js"></script>
  <![endif]-->  <link rel="stylesheet" href = "assets/css/ribbons.css">

</head>
<body>
  <div class="reveal">
    <div class="slides">
      <section class='' data-state='' id='slide-1'>
  <h2>Sequence Alignment Pipeline</h2>
  <p><br></br>
<strong>Mukarram Hossain</strong><br>
Department of Veterinary Medicine<br>
University of Cambridge</p>

<p><br></br></p>

<p><img src="assets/img/uc-colour.png" width="30%" style="border: 0px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-2'>
  <h3>Multiple sequence alignment</h3>
  <ul>
<li>Evolutionary studies typically start by aligning molecular sequences</li>
<li>Multiple sequence alignment (MSA) represents the shared evolutionary history among a group of sequences</li>
<li>Sequences are aligned by establishing sitewise homology among related residues</li>
<li>Alignments inform us about:

<ul>
<li>evolutionary process</li>
<li>structural/functional diversity</li>
<li>pattern of selection</li>
</ul></li>
</ul>

</section>
<section class='' data-state='' id='slide-3'>
  <h3>Sequence alignment</h3>
  <ul>
<li>Computational tools are used to automatically align set of sequences</li>
<li>Residues (e.g. A, T, C, G) placed together in columns are believed to have evolved from a common ancestor</li>
<li>Mismatch between residues represents substitution</li>
<li>Gaps (&#39;-&#39;) in the alignment usually denote insertion/deletion</li>
</ul>

<p><img src="assets/img/aln.png" width="100%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-4'>
  <h3>Do we need a new aligner?</h3>
  <ul>
<li>Majority of the aligners are developed for mammalian sequences</li>
<li>Viruses evolve differently than mammals:

<ul>
<li>high mutation rate</li>
<li>fast generation time</li>
<li>error prone replication</li>
<li>higher indel rate</li>
<li>frequent recombination</li>
</ul></li>
<li>Viruses are &#39;measurably evolving&#39; </li>
</ul>

</section>
<section class='' data-state='' id='slide-5'>
  <h3>Do we need a new aligner?</h3>
  <ul>
<li>Aligners often fail to produce accurate alignment of virus sequences</li>
<li>Sequence data may contain whole genome, complete genes or short fragments</li>
<li>Unreliable sequences can be present</li>
</ul>

</section>
<section class='' data-state='' id='slide-6'>
  <h3>Alignment Pipeline (Pipelign)</h3>
  <ul>
<li>We have developed a semi-automatic alignment pipeline for virus sequences</li>
<li>Pipelign progressively aligns groups of sequences</li>
<li>Systematically identifies potential unrelated fragments</li>
<li>Cluster of alignments are merged together </li>
</ul>

</section>
<section class='' data-state='' id='slide-7'>
  <h3>Pipelign - split</h3>
  <p><br></br><br></br></p>

<p><img src="assets/img/split.png" width="100%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-8'>
  <h3>Pipelign - Clustering</h3>
  <p><br></br><br></br></p>

<p><img src="assets/img/cdhit.png" width="100%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-9'>
  <h3>Pipelign - Align &amp; HMM</h3>
  <p><img src="assets/img/alignhmm.png" width="40%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-10'>
  <h3>Pipelign - HMM database</h3>
  <p><img src="assets/img/hmmdb.png" width="70%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-11'>
  <h3>Pipelign - Cluster fragments</h3>
  <p><br></br></p>

<p><img src="assets/img/fragsearch.png" width="100%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-12'>
  <h3>Pipelign - Align fragments</h3>
  <p><img src="assets/img/fragalign.png" width="60%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-13'>
  <h3>Pipelign - Merge clusters</h3>
  <p><img src="assets/img/merge.png" width="70%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-14'>
  <h3>Benchmarking</h3>
  <ul>
<li>Process of evaluating by comparison with a standard</li>
<li>Alignment tools are usually benchmarked using mammalian structural protein databases:

<ul>
<li>BAliBASE</li>
<li>PREFAB</li>
<li>HOMSTRAD</li>
<li>Simulation</li>
</ul></li>
<li>No standard for virus sequence alignments</li>
</ul>

</section>
<section class='' data-state='' id='slide-15'>
  <h3>Benchmarking - Metrices</h3>
  <ul>
<li>Alignments are generated by different methods</li>
<li>Compared with the &#39;TRUE&#39; structural alignment</li>
<li>Accuracy is measured by calculating percentage of alignment sites matching/mismathing the &#39;TRUE&#39;  alignment</li>
<li>MetAl, FastSP</li>
</ul>

</section>
<section class='' data-state='' id='slide-16'>
  <h3>BAliBASE</h3>
  <ul>
<li>Developed and managed by Julie Thompson et al.</li>
<li>Contains 218 full length and 168 homologous sequence sets and their TRUE alignments</li>
<li>Divided into 6 groups</li>
</ul>

</section>
<section class='' data-state='' id='slide-17'>
  <h3>BAliBASE - MetAl</h3>
  <p><br></br></p>

<p><img src="assets/img/balibase.metal.table.png" width="80%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-18'>
  <h3>BAliBASE - MetAl</h3>
  <p><img src="assets/img/balibase.metal.png" width="70%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-19'>
  <h3>BAliBASE - SP</h3>
  <p><img src="assets/img/balibase.sp.png" width="70%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-20'>
  <h3>PREFAB</h3>
  <ul>
<li>Developed by Robert Edgar</li>
<li>Was used to benchmark MUSCLE</li>
<li>Contains 1682 alignments</li>
</ul>

</section>
<section class='' data-state='' id='slide-21'>
  <h3>PREFAB - MetAl</h3>
  <p><img src="assets/img/prefab.metal.png" width="70%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-22'>
  <h3>PREFAB - SP</h3>
  <p><img src="assets/img/prefab.sp.png" width="70%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-23'>
  <h3>Simulation</h3>
  <ul>
<li>Two datasets are simulated using INDELible:

<ul>
<li>DENV</li>
<li>Influenza HA gene</li>
</ul></li>
<li>Each datasets contain 20 replicates of 100 sequences </li>
<li>Simulation starts with a 1000bp long root sequence</li>
<li>Substitutions, insertions and deletions are applied</li>
</ul>

</section>
<section class='' data-state='' id='slide-24'>
  <h3>DENV - MetAl</h3>
  <p><img src="assets/img/denv.metal.png" width="70%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-25'>
  <h3>DENV - SP</h3>
  <p><img src="assets/img/denv.sp.png" width="70%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-26'>
  <h3>Influenza - MetAl</h3>
  <p><img src="assets/img/flu.metal.png" width="70%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-27'>
  <h3>Influenza - SP</h3>
  <p><img src="assets/img/flu.sp.png" width="70%" style="margin: 10px 10px">&nbsp;</p>

</section>
<section class='' data-state='' id='slide-28'>
  <h3>Work in Progress</h3>
  <ul>
<li>PANGEA simulated sequences</li>
<li>Work with ambiguous sequences</li>
<li>Finalise the pipeline</li>
<li>Develope Docker container and Biobox container</li>
</ul>

</section>
<section class='' data-state='' id='slide-29'>
  
  <p><br><br></br></br>
<img src="assets/img/questions.jpg" width="400px" class="centred" style="margin: 10px 10px" /></p>

</section>
    </div>
  </div>
</body>
  <script src="libraries/frameworks/revealjs/lib/js/head.min.js"></script>
  <script src="libraries/frameworks/revealjs/js/reveal.min.js"></script>
  <script>
  // Full list of configuration options available here:
  // https://github.com/hakimel/reveal.js#configuration
  Reveal.initialize({
    controls: true,
    progress: true,
    history: true,
    center: false,
    theme: Reveal.getQueryHash().theme || 'serif', 
    transition: Reveal.getQueryHash().transition || 'cube', 
    dependencies: [
    // Cross-browser shim that fully implements classList -
    // https://github.com/eligrey/classList.js/
      { src: 'libraries/frameworks/revealjs/lib/js/classList.js', condition: function() { return !document.body.classList;}},
      // Zoom in and out with Alt+click
      { src: 'libraries/frameworks/revealjs/plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
      // Speaker notes
      { src: 'libraries/frameworks/revealjs/plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } },
      // Remote control your reveal.js presentation using a touch device
      //{ src: 'libraries/frameworks/revealjs/plugin/remotes/remotes.js', async: true, condition: function() { return !!document.body.classList; } }
      ]
  });
  </script>  <!-- LOAD HIGHLIGHTER JS FILES -->
<script src="libraries/highlighters/highlight.js/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<!-- DONE LOADING HIGHLIGHTER JS FILES -->
 

</html>