---
title       : Sequence Alignment Pipeline 
subtitle    : 
author      : Mukarram Hossain and Simon Frost
job         : University of Cambridge
framework   : revealjs        # {io2012, html5slides, shower, dzslides, ...}
highlighter : highlight.js  # {highlight.js, prettify, highlight}
hitheme     : tomorrow      # 
widgets     : []            # {mathjax, quiz, bootstrap}
mode        : selfcontained # {standalone, draft}
revealjs:
  theme: serif
  transition: cube
  center: "false"
knit        : slidify::knit2slides
---

## Sequence Alignment Pipeline
<br></br>
**Mukarram Hossain**  
Department of Veterinary Medicine  
University of Cambridge
  
<br></br>

<img src="assets/img/uc-colour.png" width="30%" style="border: 0px">&nbsp;


---

## Multiple sequence alignment

- Evolutionary studies typically start by aligning molecular sequences
- Multiple sequence alignment (MSA) represents the shared evolutionary history among a group of sequences
- Sequences are aligned by establishing sitewise homology among related residues
- Alignments inform us about:
    - evolutionary process
    - structural/functional diversity
    - pattern of selection
    - and many others

---

## Sequence alignment

- Computational tools are used to automatically align set of sequences
- Residues (e.g. A, T, C, G) placed together in columns are believed to have evolved from a common ancestor
- Mismatch between residues represents substitution
- Gaps ('-') in the alignment usually denote insertion/deletion

<img src="assets/img/aln.png" width="100%" style="margin: 10px 10px">&nbsp;

---

## Do we need a new aligner?

- Majority of the aligners are developed for mammalian sequences
- Viruses evolve differently than mammals:
    - high mutation rate
    - fast generation time
    - error prone replication
    - higher indel rate
    - frequent recombination
- Viruses are 'measurably evolving' 

---

## Do we need a new aligner?

<br></br>

- Aligners often fail to produce accurate alignment of virus sequences
- Sequence data may contain whole genome, complete genes or short fragments
- Unreliable sequences can be present

---

## Alignment Pipeline (Pipelign)

<br></br>

- We have developed a semi-automatic alignment pipeline for virus sequences
- Pipelign progressively aligns groups of sequences
- Systematically identifies potential unrelated fragments
- Cluster of alignments are merged together 

---

## Pipelign - split 

<br></br><br></br>

<img src="assets/img/split.png" width="100%" style="margin: 10px 10px">&nbsp;

---

## Pipelign - Clustering

<br></br><br></br>

<img src="assets/img/cdhit.png" width="100%" style="margin: 10px 10px">&nbsp;


---

## Pipelign - Align & HMM

<img src="assets/img/alignhmm.png" width="40%" style="margin: 10px 10px">&nbsp;


---

## Pipelign - HMM database

<img src="assets/img/hmmdb.png" width="70%" style="margin: 10px 10px">&nbsp;

---

## Pipelign - Cluster fragments

<br></br>

<img src="assets/img/fragsearch.png" width="100%" style="margin: 10px 10px">&nbsp;

---

## Pipelign - Align fragments

<img src="assets/img/fragalign.png" width="60%" style="margin: 10px 10px">&nbsp;

---

## Pipelign - Merge clusters

<img src="assets/img/merge.png" width="70%" style="margin: 10px 10px">&nbsp;

---

## Benchmarking

<br></br>

- Process of evaluating by comparison with a standard
- Alignment tools are usually benchmarked using mammalian structural protein databases:
    - BAliBASE
    - PREFAB
    - HOMSTRAD
    - Simulation
- No standard for virus sequence alignments

---

## Benchmarking - Metrices

<br></br>

- Alignments are generated by different methods
- Compared with the 'TRUE' structural alignment
- Accuracy is measured by calculating percentage of alignment sites matching/mismathing the 'TRUE'  alignment
- MetAl, FastSP

---

## BAliBASE

<br></br>

- Developed and managed by Julie Thompson et al.
- Contains 218 full length and 168 homologous sequence sets and their TRUE alignments
- Divided into 6 groups

---

## BAliBASE - MetAl

<br></br>

<img src="assets/img/balibase.metal.table.png" width="80%" style="margin: 10px 10px">&nbsp;

---

## BAliBASE - MetAl


<img src="assets/img/balibase.metal.png" width="70%" style="margin: 10px 10px">&nbsp;


---

## BAliBASE - SP


<img src="assets/img/balibase.sp.png" width="70%" style="margin: 10px 10px">&nbsp;

---

## PREFAB

<br></br>

- Developed by Robert Edgar
- Was used to benchmark MUSCLE
- Contains 1682 alignments

---

## PREFAB - MetAl

<img src="assets/img/prefab.metal.png" width="70%" style="margin: 10px 10px">&nbsp;

---

## PREFAB - SP

<img src="assets/img/prefab.sp.png" width="70%" style="margin: 10px 10px">&nbsp;

---

## Simulation

<br></br>

- Two datasets are simulated using INDELible:
    - DENV
    - Influenza HA gene
- Each datasets contain 20 replicates of 100 sequences 
- Simulation starts with a 1000bp long root sequence
- Substitutions, insertions and deletions are applied

---

## DENV - MetAl

<img src="assets/img/denv.metal.png" width="70%" style="margin: 10px 10px">&nbsp;

---

## DENV - SP

<img src="assets/img/denv.sp.png" width="70%" style="margin: 10px 10px">&nbsp;

---

## Influenza - MetAl

<img src="assets/img/flu.metal.png" width="70%" style="margin: 10px 10px">&nbsp;

---

## Influenza - SP

<img src="assets/img/flu.sp.png" width="70%" style="margin: 10px 10px">&nbsp;

---

## Work in Progress

<br></br>

- PANGEA simulated sequences
- Work with ambiguous sequences
- Finalise the pipeline
- Develope Docker container and Biobox container

---

<br><br></br></br>
<img src="assets/img/questions.jpg" width="400px" class="centred" style="margin: 10px 10px" />